js/everything.js

```diff
@@ -206,11 +212,55 @@ var ZenPlayer = {
                     seconds: plyrPlayer.plyr.embed.getDuration(),
                     youtubeID: plyrPlayer.plyr.embed.getVideoData().video_id
                 });
-
                 // Show player
                 that.show();
                 updateTweetMessage();
             });
+            // when player has finished playing
+            plyrPlayer.addEventListener("ended", function() {
+                plyrPlayer.removeEventListener("ended");
+                videoMetaData = window.sessionStorage.getItem("videoMetaData");
+                videoMetaData = JSON.parse(videoMetaData);
+                if (videoMetaData !== null) {
+                    videoMetaData["items"].push({"type" : "youtube", "id" : currentVideoID, "played" : "true" });
+                    videoMetaData["autoplayState"] = autoplayState ;
+                    if (autoplayState !== null && autoplayState === "true") {
+                        /*
+                        videoMetaData contains two variables, an autoplay state and information (list) about all videos played till now.
+                        It is stored in session storage, and retrieved when relevant entries need to be added/modified.
+                        */
+                        // if playList songs left
+                        if (playList.length !== 0)
```

Comment:
The first one seems better I guess? In the second one, there's only a check with the first entry for duplicates. If nextID differs from the first ID, its simply played and not checked with the remaining entries. That sounds wrong